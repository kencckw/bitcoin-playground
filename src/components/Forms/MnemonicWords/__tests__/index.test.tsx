import React from 'react';
import renderer from 'react-test-renderer';
import { MnemonicWordsForm as Component } from '../';

it('renders correctly', () => {
  const tree = renderer.create(<Component />).toJSON();
  expect(tree).toMatchSnapshot();
});
