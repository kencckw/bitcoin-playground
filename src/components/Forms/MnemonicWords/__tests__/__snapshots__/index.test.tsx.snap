// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-row ant-row-center ant-row-top"
  style={
    Object {
      "paddingTop": 12,
    }
  }
>
  <div
    className="ant-col ant-col-sm-24 ant-col-lg-16 ant-col-xl-12"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-card ant-card-bordered"
      style={
        Object {
          "minHeight": 650,
        }
      }
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <form
          className="ant-form ant-form-horizontal"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="ant-page-header ant-page-header-ghost"
            style={
              Object {
                "paddingLeft": 0,
              }
            }
          >
            <div
              className="ant-page-header-heading"
            >
              <div
                className="ant-page-header-heading-left"
              >
                <span
                  className="ant-page-header-heading-title"
                  title="Mnemonic words"
                >
                  Mnemonic words
                </span>
                <span
                  className="ant-page-header-heading-sub-title"
                  title="Import/generate a mnemonic words"
                >
                  Import/generate a mnemonic words
                </span>
              </div>
              <span
                className="ant-page-header-heading-extra"
              >
                <button
                  className="ant-btn ant-btn-primary ant-btn-background-ghost"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    aria-label="reload"
                    className="anticon anticon-reload"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="reload"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                      />
                    </svg>
                  </span>
                  <span>
                    Random
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="size"
                title="Number of words"
              >
                Number of words
              </label>
            </div>
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <div
                    className="ant-select ant-select-single ant-select-show-arrow"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                  >
                    <div
                      className="ant-select-selector"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <span
                        className="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="size_list_0"
                          aria-autocomplete="list"
                          aria-controls="size_list"
                          aria-haspopup="listbox"
                          aria-owns="size_list"
                          autoComplete="off"
                          className="ant-select-selection-search-input"
                          id="size"
                          onChange={[Function]}
                          onCompositionEnd={[Function]}
                          onCompositionStart={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onPaste={[Function]}
                          readOnly={true}
                          role="combobox"
                          style={
                            Object {
                              "opacity": 0,
                            }
                          }
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        className="ant-select-selection-placeholder"
                      />
                    </div>
                    <span
                      aria-hidden={true}
                      className="ant-select-arrow"
                      onMouseDown={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        className="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="language"
                title="Language"
              >
                Language
              </label>
            </div>
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <div
                    className="ant-select ant-select-single ant-select-show-arrow"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                  >
                    <div
                      className="ant-select-selector"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                    >
                      <span
                        className="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="language_list_0"
                          aria-autocomplete="list"
                          aria-controls="language_list"
                          aria-haspopup="listbox"
                          aria-owns="language_list"
                          autoComplete="off"
                          className="ant-select-selection-search-input"
                          id="language"
                          onChange={[Function]}
                          onCompositionEnd={[Function]}
                          onCompositionStart={[Function]}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onPaste={[Function]}
                          readOnly={true}
                          role="combobox"
                          style={
                            Object {
                              "opacity": 0,
                            }
                          }
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        className="ant-select-selection-placeholder"
                      />
                    </div>
                    <span
                      aria-hidden={true}
                      className="ant-select-arrow"
                      onMouseDown={[Function]}
                      style={
                        Object {
                          "WebkitUserSelect": "none",
                          "userSelect": "none",
                        }
                      }
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        className="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="passphrase"
                title="Passphrase"
              >
                Passphrase
              </label>
            </div>
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <span
                    className="ant-input-affix-wrapper ant-input-password"
                    onMouseUp={[Function]}
                  >
                    <input
                      action="click"
                      autoComplete="off"
                      className="ant-input"
                      id="passphrase"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      style={null}
                      type="password"
                      value=""
                    />
                    <span
                      className="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        className="anticon anticon-eye-invisible ant-input-password-icon"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onMouseUp={[Function]}
                        role="img"
                        tabIndex={-1}
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <textarea
                    className="ant-input"
                    id="value"
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    style={Object {}}
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-page-header ant-page-header-ghost"
            style={
              Object {
                "paddingLeft": 0,
              }
            }
          >
            <div
              className="ant-page-header-heading"
            >
              <div
                className="ant-page-header-heading-left"
              >
                <span
                  className="ant-page-header-heading-title"
                  title="Seed"
                >
                  Seed
                </span>
                <span
                  className="ant-page-header-heading-sub-title"
                  title="Calculated from the mnemonic words"
                >
                  Calculated from the mnemonic words
                </span>
              </div>
              <span
                className="ant-page-header-heading-extra"
              >
                <button
                  className="ant-btn ant-btn-primary ant-btn-background-ghost"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    aria-label="copy"
                    className="anticon anticon-copy"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="copy"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                      />
                    </svg>
                  </span>
                  <span>
                    Copy
                  </span>
                </button>
              </span>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <textarea
                    autoComplete="off"
                    className="ant-input ant-input-disabled"
                    disabled={true}
                    id="seed"
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    style={Object {}}
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
